### PR TITLE
Fixes in wave structure, solves #1182

### LINKED ISSUE #1182: inconsistent array copies in MOM_wave_structure
@Hallberg-NOAA I know you're looking at this rn but just wanted to point out this problem:

when layers are merged, this becomes incorrect:

https://github.com/NOAA-GFDL/MOM6/blob/dd75807adddb749aecee20cf6a992904269b5900/src/diagnostics/MOM_wave_structure.F90#L488

also similar issues on line 521 and below.
I can submit a PR unless it would interfere with your rewriting

### PR DESCRIPTION
**Pull request by raphaeldussin**
fix #1182 

* condition on number of layers does not meet requirements of 3 layers minimum for first mode.
* array copies are inconsistent in size
* allow to run internal_tides code without a specified input file (using default values) so code can be
exercised in test case (e.g Baltic)

### CODE REVIEW

File: src/parameterizations/lateral/MOM_internal_tides.F90 [hunk 1]
```diff
@@ -2324,12 +2324,14 @@ subroutine internal_tides_init(Time, G, GV, US, param_file, diag, CS)
...
                "The path to the file containing the local angle of "//&
                "the coastline/ridge/shelf with respect to the equator.", &
-               fail_if_missing=.false.)
+               fail_if_missing=.false., default='')
   filename = trim(CS%inputdir) // trim(refl_angle_file)
   call log_param(param_file, mdl, "INPUTDIR/REFL_ANGLE_FILE", filename)
   allocate(CS%refl_angle(isd:ied,jsd:jed)) ; CS%refl_angle(:,:) = CS%nullangle
-  call MOM_read_data(filename, 'refl_angle', CS%refl_angle, &
-                     G%domain, timelevel=1)
+  if (file_exists(filename, G%domain)) then
```
Comment (Hallberg-NOAA) [inline src/parameterizations/lateral/MOM_internal_tides.F90:2331]: I think that we need to think about how to handle the case where the input file does not exist both here and in the other two changes in this file. If the file is not specified (i.e., the default of ' ') silently not doing anything is probably the right answer, but if a name is given but the file does not exist there should either be a warning message or a fatal error.

Comment (raphaeldussin) [reply src/parameterizations/lateral/MOM_internal_tides.F90:2330]: Agreed and fixed in new commit, also move the call to log_param. No need to log an empty parameter, right?

**Comment by raphaeldussin** [comment]
looks like travis froze

**Comment by raphaeldussin** [comment]
reboot travis

**Comment by marshallward** [comment]
Gaea regression test: https://gitlab.gfdl.noaa.gov/ogrp/MOM6/-/pipelines/11054

**Pull request merged**